src/hb-ot-var-common.hh

```diff
@@ -1031,10 +1031,13 @@ struct TupleVariationData
 {
   bool sanitize (hb_sanitize_context_t *c) const
   {
-    TRACE_SANITIZE (this);
-    // here check on min_size only, TupleVariationHeader and var data will be
+    // here check on min_size and TupleVariationHeader array only, var data will be
     // checked while accessing through iterator.
-    return_trace (c->check_struct (this));
+    TRACE_SANITIZE (this);
+    if (!(c->check_struct (this)))
+        return_trace (false);
+
+    return_trace (c->check_range (this, min_size + TupleVariationHeader::min_size * (tupleVarCount.get_count ())));
```

Comment:
Should that be `static_size` instead of `min_size`?

Reply:
Humm. I'm not sure about this change. Why is it needed again?

Reply:
when you init tuple iterator here: https://github.com/harfbuzz/harfbuzz/blob/main/src/hb-ot-var-common.hh#L1525, this pointer is out of range already: https://github.com/harfbuzz/harfbuzz/blob/main/src/hb-ot-var-gvar-table.hh#L526,  I thought we should at least check TupleVariationHeader arrays' min_size.
TupleVariationHeader has no static_size, since we might or might not have peak/intermediate tuples.

Reply:
or  I can add a sanitize() for each TupleVariationHeader so we check the exact size of each header

Reply:
Oh you're right, we don't need this. I should get var_data from hb_blob, which has total length info so we won't access any pointer beyond that. I'll revert this commit and make another fix